src/backend/direct.rs

```diff
@@ -546,7 +546,12 @@ impl crate::Context for Context {
         trace_dir: Option<&std::path::Path>,
     ) -> Self::RequestDeviceFuture {
         let global = &self.0;
-        let device_id = wgc::gfx_select!(*adapter => global.adapter_request_device(*adapter, desc, trace_dir, PhantomData)).unwrap();
+        let device_id = match wgc::gfx_select!(
+            *adapter => global.adapter_request_device(*adapter, desc, trace_dir, PhantomData)
+        ) {
+            Ok(d) => d,
+            Err(e) => panic!("{}", e),
+        };
```

Comment:
Since this pattern is repeated in a lot of places, would it be possible to use a generic function? Or use a macro?

Reply:
Discussed in the matrix that a function like `fn on_error(err: impl Error) { panic!("{}", err) )` and calling `.unwrap_or_else(on_error)` is the desired way to go.